Documentation/api_rfc/v2_acl_api.md

```diff
@@ -0,0 +1,253 @@
+
+### Role
+Each role has exact one associated ACL. An access control list specifies the capacity on key-value resources. A role with manage capacity of a key-value resource can grant/revoke capacity of that key-value to other roles.
+
+The special static ROOT role has a full manage capacity on all key-value resources, the capacity to manage user resources and setting resources. Only the ROOT role has the capacity to manage user resources.
+
+### Key-Value
+A key-value resource is a set of key-value pairs in the key-value store. The ACL specifies the capacity of a key-value resource (a set of key-value pairs) not of one key-value pairs in the key-value store. If an ACL has the read capacity on /foo, it has the read capacity on all key-value pairs which has a key prefixed with /foo. 
+
+## v2 Auth
+
+### Basic Auth
+We only support Basic Auth for the first version. Client needs to attach the basic auth to the HTTP Authorization Header. 
+http://en.wikipedia.org/wiki/Basic_access_authentication
+
+### Authorization field for operations
+Added to requests to /v2/keys, /v2/acl
+Add code 403 Forbidden to the set of responses from the v2 API
+Authorization: Basic {encoded string}
+
+## v2 ACL
+Only ”allow” prefixes (incidentally, this is what Amazon S3 does)
+if /a is allowed in v2, then /abcd is allowed.
+Options could include the control list specifically stating the prefix, eg:
+/v2/keys/……
+/v2/manage/….
+/v3/keys/…..
+
+Deny prefixes TBD
+
+### API endpoints
+
+#### Users
+
+#####Get a list of users
+
+GET/HEAD  /v2/acl/user
+
+        Sent Headers:
+	Authorization: Basic <BasicAuthString>
+        Possible Status Codes:
+            200 OK
+            403 Forbidden
+        200 Headers:
+            ETag: "<hash of list of users>”
+	Content-type: application/json
+        200 Body:
+            {
+  		“users”: [“alice”, “bob”, “eve”]
+            }
+
+##### Get User Details
+GET/HEAD  /v2/acl/user/alice
+
+        Sent Headers:
+	Authorization: Basic <BasicAuthString>
+        Possible Status Codes:
+            200 OK
+            403 Forbidden
+	404 Not Found
+        200 Headers:
+            ETag: "user/alice:<lastModified>"
+	Content-type: application/json
+        200 Body:
+            {
+  		“user” : “alice”
+		“roles” : [“fleet”, “etcd”]
+		“lastModified”: “2015-02-05Z18:00:00”
+            }
+
+##### Create A User
+PUT  /v2/acl/user/charlie
+
+        Sent Headers:
+	Authorization: Basic <BasicAuthString>
+        Put Body:
+charliePassHunter2
+        Possible Status Codes:
+            200 OK
+            403 Forbidden
+        200 Headers:
+            ETag: "user/charlie:<tzNow>"
+        200 Body: (empty)
+
+##### Remove A User
+DELETE  /v2/acl/user/charlie
+        Sent Headers:
+	Authorization: Basic <BasicAuthString>
+        Possible Status Codes:
+            200 OK
+            403 Forbidden
+        200 Headers:
+        200 Body:
+
+##### Grant a Role(s) to a User
+PUT  /v2/acl/user/charlie/grant
```

Comment:
it is not Restful to put action in the url. why don't we just use PUT for grant and use DELETE for revoke?